### ISSUE TITLE
A function with two datetime parameters receives the string '00:00:00.000000' for the second argument

### ISSUE THREAD

**Issue by gthb** [oguimbal/pg-mem#236]
# Describe the bug

Register a function with two parameters, both of type `DataType.time`. The second argument passed to the function implementation is never the argument passed in the SQL, it is always the string '00:00:00.000000'.

# To Reproduce

This test fails (I poked it into `function-calls.spec.ts`):
```js
    it('passes multiple arguments', () => {
        db.public.registerFunction({
            name: 'datediff',
            args: [DataType.time, DataType.time],
            returns: DataType.integer,
            implementation: (arg1, arg2) => {
                assert.instanceOf(arg1, Date);
                assert.instanceOf(arg2, Date);
                return Math.trunc(arg1 / 86400) - Math.trunc(arg2 / 86400);
            },
        });

        expect(many(`select datediff(date '2008-03-13', date '2008-02-12')`))
            .to.equal(29);
    });
```
with the assertion failure:
```
  1) Functions
       passes multiple arguments:
     AssertionError: expected '00:00:00.000000' to be an instance of Date
```

# pg-mem version

2.4.3 (and tip of master branch)

**Comment by oguimbal** [comment]
`time` type is not a date, it only is a time of day, for which Javascript has no obvious primitive that can be used => I chose to pass them around as `HH:mm:ss.xxxxx` strings.
Hi !

For actual dates, use `timestamp`  or `date` types

![image](https://user-images.githubusercontent.com/8973947/172441427-f8408609-9326-4b01-876d-b894d335e703.png)

**Comment by oguimbal** [comment]
Well, actualy, `date` is not supposed to be castable to `time` 😄 so your query should have failed.

I'll fix that. But as for your issue, my comment holds

**Comment by oguimbal** [comment]
The issue with cast should be fixed in `pg-mem@2.5.0`
